torch/_parameter.py

```diff
@@ -0,0 +1,244 @@
+"""
+The :mod:`torch.nn.parameter` module provides a kind of :class:`~torch.Tensor`
+that is to be considered a module parameter.
+
+The class definitions need to be here instead of inside ``torch/nn/`` so they
+can be used in other places in ``torch/`` (namely :mod:`torch.autograd`) without
+running into circular dependency problems
+"""
```

Comment:
> Sure but I don't see why we do it here. Parameter() is 100% tied to the concept of Module. You could say the same that Module doesn't depend on nn and so could be moved out.

I have added a description of why we need this.

------

Also, recap for the PR description.

This change is used to break the circular dependency between `torch.autograd` (requires `torch.nn.paramter.Parameter`) and `torch.nn` (requires `torch.autograd.grad_mode.no_grad`).

https://github.com/pytorch/pytorch/blob/a5f816df18f49619cfb15ffcd3b74606a495c4e2/torch/csrc/autograd/init.cpp#L123-L131

Reply:
I'm not sure what you mean by this. These things work today, so we don't actually have a problematic dependency cycle?